plugins/module_utils/foreman_helper.py

```diff
@@ -504,6 +514,13 @@ def _flatten_entity(entity, entity_spec):
     return result
 
 
+# Helper to minimize API calls to retrive smart proxies
+def get_smart_proxy(module, smart_proxies, smart_proxy_name):
+    if smart_proxy_name not in smart_proxies:
+        smart_proxies[smart_proxy_name] = module.find_resource_by_name('smart_proxies', name=smart_proxy_name, failsafe=False, thin=True)
+    return smart_proxies[smart_proxy_name]
+
```

Comment:
I like the effort to not search for things twice. But i think, we could have this in a generic way by passing a memory / cache / you-name-it dictionary to the `find_resource` which will be looked into first and will receive the result.
It would, almost transparently, solve the find my organizations problem, too.

Reply:
:+1: on Matthias idea, if anywhere this should be directly in the generic find helper.

Reply:
I think it's ok to use the `search` string as the key in the memory dict.